sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java

```diff
@@ -45,27 +46,33 @@
     static HttpPipeline buildPipeline(
         TablesSharedKeyCredential tablesSharedKeyCredential,
         TokenCredential tokenCredential, AzureSasCredential azureSasCredential, String sasToken,
-        String endpoint, RequestRetryOptions retryOptions, HttpLogOptions logOptions,
-        HttpClient httpClient, List<HttpPipelinePolicy> additionalPolicies,
-        Configuration configuration, ClientLogger logger) {
+        String endpoint, RequestRetryOptions retryOptions, HttpLogOptions logOptions, ClientOptions clientOptions,
+        HttpClient httpClient, List<HttpPipelinePolicy> perCallAdditionalPolicies,
+        List<HttpPipelinePolicy> perRetryAdditionalPolicies, Configuration configuration, ClientLogger logger) {
+
+        configuration = (configuration == null) ? Configuration.getGlobalConfiguration() : configuration;
+        clientOptions = (clientOptions == null) ? new ClientOptions() : clientOptions;
+        logOptions = (logOptions == null) ? new HttpLogOptions() : logOptions;
 
         validateSingleCredentialIsPresent(
             tablesSharedKeyCredential, tokenCredential, azureSasCredential, sasToken, logger);
 
-        //1
+        // Closest to API goes first, closest to wire goes last.
         List<HttpPipelinePolicy> policies = new ArrayList<>();
-        policies.add(getUserAgentPolicy(configuration));
+        policies.add(getUserAgentPolicy(configuration, clientOptions, logOptions));
         policies.add(new RequestIdPolicy());
 
         // Add Accept header so we don't get back XML.
         // Can be removed when this is fixed. https://github.com/Azure/autorest.modelerfour/issues/324
```

Comment:
The issue mentioned here doesn't exist anymore, is this still an issue? Also, almost all Storage APIs return XML, is this building a Cosmos based client?

Reply:
The Tables clients can work with either the Storage and Cosmos APIs, which would not return XML. This was written before I joined the Tables v-team, but I suspect it's so that we only handle JSON responses no matter where they come from. I will talk to the team about this decision and change the code if needed.

Reply:
It appears adding this header is no longer necessary, will remove it.

Reply:
Removing this header messes with our deserialization for some scenarios, I will leave it for now and investigate if it can be removed by making our `JacksonAdapter` more robust.